src/components/ec-timer/ec-timer.vue

```diff
@@ -93,43 +96,38 @@ export default {
     offset() {
       return this.circumference + this.steps * this.secondsLeft;
     },
-    timeDifference() {
-      return this.currentTime - this.startTime;
-    },
   },
   watch: {
     isRunning: {
       immediate: true,
       handler(value) {
         if (value) {
-          this.countdown();
+          this.startCountdown();
         } else {
-          clearInterval(this.timerInterval);
+          this.countdown.stop();
           this.secondsLeft = this.seconds;
         }
       },
     },
   },
   beforeDestroy() {
-    clearInterval(this.timerInterval);
+    this.countdown.stop();
+    this.countdown = null;
   },
   methods: {
-    countdown() {
-      this.startTime = Math.ceil(Date.now() / 1000);
-      this.timerInterval = setInterval(this.reduceSecondsLeft, 1000);
-    },
-    reduceSecondsLeft() {
-      this.currentTime = Math.ceil(Date.now() / 1000);
-      this.secondsLeft = this.seconds - this.timeDifference;
-      if (this.timeDifference >= this.seconds) {
-        clearInterval(this.timerInterval);
+    startCountdown() {
+      this.countdown.start(this.seconds);
+      this.countdown.emitter.on('time-updated', (secondsLeft) => {
```

Comment:
These events are never being deregistered, so if I call `this.startCountdown();` twice, all events are fired two times.

I think you had it right when you was initialising countdown in this function, it was preventing this behaviour to happen. If you are afraid of memory leaks, then stop previous countdown first when starting the new one:
```
startCountdown() {
  if (this.countdown) {
    this.countdown.stop();
  }
  this.countdown = new Countdown();
  this.countdown.start(this.seconds);
}
```

Reply:
Cool! done!